core/src/main/kotlin/org/evomaster/core/problem/graphql/IntrospectiveQuery.kt

```diff
@@ -61,35 +63,38 @@ class IntrospectiveQuery {
                     }
                 """.trimIndent(), MediaType.APPLICATION_JSON_TYPE)
-
-        //TODO check if TCP problems
-        val response = SearchTimeController.measureTimeMillis({ ms, res ->
+        val response: Response?
+        try {
+            response = SearchTimeController.measureTimeMillis({ ms, res ->
                 LoggingUtil.getInfoLogger().info("Fetched GraphQL schema in ${ms}ms")
-        }, {
-            try {
-                var request = client.target(graphQlEndpoint)
-                    .request("application/json")
+            }, {
+                try {
+                    var request = client.target(graphQlEndpoint)
+                            .request("application/json")
 
-                for (h in list) {
-                    request = request.header(h.first, h.second)
+                    for (h in list) {
+                        request = request.header(h.first, h.second)
+                    }
+                    request.buildPost(query)
+                            .invoke()
+                } catch (e: Exception) {
+                    log.error("Failed query to '$graphQlEndpoint' :  $query")
+                    throw e
                 }
-                request.buildPost(query)
-                    .invoke()
-            } catch (e: Exception) {
-                log.error("Failed query to '$graphQlEndpoint' :  $query")
-                throw e
-            }
-        })
+            })
+        } catch (e: Exception) {
+            throw SutProblemException("Failed introspection query to '$graphQlEndpoint', please check connection and URL format")
+        }
 
 
 
 
         /*
            Extract the body from response as a string
         */
-        val body = response.readEntity(String::class.java)
+        val body = response?.readEntity(String::class.java)
```

Comment:
not needed `?` if variable is made non-nullable, with explicit `?:` check on nullability (if needed)